LuaEngine.cpp

```diff
@@ -1008,9 +1002,6 @@ void Eluna::CleanUpStack(int number_of_arguments)
 
     lua_pop(L, number_of_arguments + 1); // Add 1 because the caller doesn't know about `event_id`.
     // Stack: (empty)
-
-    if (event_level == 0)
-        InvalidateObjects();
```

Comment:
We may still want to have the old system to invalidate pushed objects at end of an event stack.

One of the issues is that if you have an aura for example `aura = player:GetAura(spellid)`, then that aura will be valid until the aura object is destroyed.
But as I understand, aura can be transferred to other maps with a player.

If we dont use some specific event to always invalidate objects, then it can happen that an aura or other object is transferred to another map and can be accessed from two lua states at the same time during map update events when we run multi-state mode eluna.

Any thoughts on this?

Reply:
Actually, you don't see this in the PR but aura handle is not invalidated when its destructor runs (which is delayed a bit, first it gets added to m_removedAuras list)  but immediately when it is unapplied from unit (see https://github.com/TrinityCore/TrinityCore/commit/a79b42bf681e211997923dbc6c191aae187aded6#diff-f3fb604c220161af7953902aecc9586825b8193413ab94fdeea4de5665d7a393R622)

Reply:
Luckily aura already has another reference it can additionally check, its WorldObject owner - so this is what I came up with 753f2eac56c1514f2d0da6e93df209ab8032c8ee

Reply:
There are also Loot functions proposed [here](https://github.com/ElunaLuaEngine/Eluna/pull/470/files#diff-b3dfd49786d0d15a9d8888807d244dda6cd297a3a472f2ec5050657caf2d48dfR1395).
I wonder how cases like this should be handled where there is a new type exposed that does not necessarily have the unique_trackable_ptr.

I guess introducing new type would always require modifying the cores first to introduce the tracking and then the type can be exposed after.

Reply:
I'm starting to think its would be better to only allow preserving references for longer for some types, not all of them (using the old callbackid for the ones we want to exclude)

Reply:
Currently the primary objects where storing the GUID and fetching the object is the norm is; players, creatures (units) and gameobjects. I'm guessing we can start with those at the very least and expand where it makes sense.

Reply:
Should be addressed by 39eed6503417de7ddc0686e4ad01798ba8c6fe63